src/axom/sina/core/Document.hpp

```diff
@@ -284,6 +298,78 @@ Document loadDocument(std::string const &path,
+ *        don't exist in the file on disk and add them. Then, for any file it finds that does exist, it will
+ *        go through its various fields and compare them. New data/curve sets/etc. will simply be added in.
+ *        For fields that already exist:
+ *         - For curves in curve sets, new values will be appended onto the end. This is useful for ex: codes that
+ *           don't store curves in memory, appending the new set of curve values each cycle will allow the file on
+ *           disk to grow as the simulation continues.
+ *        - For everything else (data, files, user_defined...), behavior depends on mergeProtocol, with the default
+ *        behavior being to overwrite the old (on-disk) field with the new value.
+ *
+ * \param hdf5FilePath the path to the HDF5 file
+ * \param new_data a vector of the new data to append
+ * \param mergeProtocol protocol for handling duplicate data/files/user_defined/etc
+ *                            1 = take the new value. 2 = keep the old value. 3 = Cancel the append.
+ *                     Protocol 1 is the conduit default behavior. Ignored entirely when skipping validation.
+ * \param skipValidation whether to skip the validation step entirely. Most useful for well-controlled cases,
+ *                       ex: a code is appending values to every timeseries every N cycles.
+ * 
+ * \return a conduit Node containing a list of any errors encountered in appending. If empty, success!
+ */
+conduit::Node appendDocumentToHDF5(const std::string &hdf5FilePath,
+                                   Document const &newData,
+                                   const int mergeProtocol = 1,
+                                   const bool skipValidation = false);
+
+/**
+ * \brief Check a node against some file handle and return a Conduit node populated with any errors that
+ *        would prevent appending. Primarily useful for testing out what may be going wrong with an append.
+ * 
+ * \param appendTo Either a conduit node or a conduit relay filehandle (prefer the latter for HDF5) representing
+ *                 the data on disk
+ * \param appendFrom A conduit node representing the document being appended to the file on disk
+ * \param mergeProtocol protocol for handling duplicate data/files/user_defined/etc. See append methods above.
+ * \return a conduit Node containing a list of any errors encountered in appending. If empty, success.
+ */
+template <typename ConduitRelayLike>
+conduit::Node validateAppendDocument(ConduitRelayLike &appendTo,
+                                     const conduit::Node &appendFrom,
+                                     const std::string &endpoint,
+                                     const int mergeProtocol,
+                                     int record_num,
+                                     // default'd because it might go away with a conduit update
+                                     const std::string &original_file_path = "");
```

Comment:
A few of the parameters are not documented: `endpoint`, `record_num`, `original_file_path`